### PR TITLE
Terraform Azure | Added SingleGW & MDS templates

### PR DESCRIPTION
**Pull request by chkp-yairra**

### CODE REVIEW

File: terraform/azure/mds-existing-vnet/README.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
+# Check Point CloudGuard IaaS Management Terraform deployment for Azure
```
Comment (chkp-dmitrygo) [inline terraform/azure/mds-existing-vnet/README.md:1]: CloudGuard Network Security, not CloudGuard IaaS

File: terraform/azure/mds-existing-vnet/README.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
+# Check Point CloudGuard Network Security MDS Terraform deployment for Azure
+
+This Terraform module deploys Check Point CloudGuard IaaS Management solution into a new Vnet in Azure.
```
Comment (chkp-natanelm) [inline terraform/azure/mds-existing-vnet/README.md:3]: CloudGuard IaaS -> CloudGuard Network Security

File: terraform/azure/mds-existing-vnet/README.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+## Configurations
+- Install and configure Terraform to provision Azure resources: [Configure Terraform for Azure](https://docs.microsoft.com/en-us/azure/virtual-machines/linux/terraform-install-configure)
+- In order to use ssh connection to VMs, it is **required** to add a public key to the /terraform/azure/mds-existing-vnet/azure_public_key file.
+
+## Usage
+- Choose the preferred login method to Azure in order to deploy the solution:
+    <br>1. Using Service Principal:
+    - Create a [Service Principal](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal) (or use the existing one) 
+    - Grant the Service Principal at least "**Owner**" permissions to the Azure subscription<br>
```
Comment (chkp-natanelm) [inline terraform/azure/mds-existing-vnet/README.md:23]: For MDS we don't need owner permission.
Please update to the most restrictive role needed.

File: terraform/azure/mds-existing-vnet/README.md [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+ 
+        terraform plan
+- Create or modify the deployment:
+ 
+        terraform apply
+
+### terraform.tfvars variables:
+ | Name          | Description   | Type          | Allowed values |
+ | ------------- | ------------- | ------------- | -------------  |
+ | **client_secret** | passwordThe client secret of the Service Principal used to deploy the solution | string |
```
Comment (chkp-natanelm) [inline terraform/azure/mds-existing-vnet/README.md:75]: passwordThe ?

File: terraform/azure/mds-existing-vnet/README.md [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+ |  |  |  |  |  |
+ | **tenant_id** | The tenant ID of the Service Principal used to deploy the solution | string |
+ |  |  |  |  |  |
+ | **subscription_id** | The subscription ID is used to pay for Azure cloud services | string |
+ |  |  |  |  |  |
+ | **source_image_vhd_uri** | The URI of the blob containing the development image. Please use noCustomUri if you want to use marketplace images. | string | 
+ |  |  |  |  |  |
+ | **resource_group_name** | The name of the resource group that will contain the contents of the deployment | string | Resource group names only allow alphanumeric characters, periods, underscores, hyphens and parenthesis and cannot end in a period |
+ |  |  |  |  |  |
+ | **mds_name** | MDS name. | string |
```
Comment (chkp-natanelm) [inline terraform/azure/mds-existing-vnet/README.md:87]: MDS name**.**

Please decide if you want to use dots or not, and align for all.

File: terraform/azure/mds-existing-vnet/README.md [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+ |  |  |  |  |  |
+ | **bootstrap_script** | An optional script to run on the initial boot | string | Bootstrap script example: <br/>"touch /home/admin/bootstrap.txt; echo 'hello_world' > /home/admin/bootstrap.txt" <br/>The script will create bootstrap.txt file in the /home/admin/ and add 'hello word' string into it |
+ |  |  |  |  |  |
+ | **allow_upload_download** | Automatically download Blade Contracts and other important data. Improve product experience by sending data to Check Point | boolean | true; <br/>false; |
+ |  |  |  |  |  |
+ | **authentication_type** | Specifies whether a password authentication or SSH Public Key authentication should be used | string | "Password"; <br/>"SSH Public Key"; |
+ |  |  |  |  |  |
+ | **admin_shell** | Enables to select different admin shells | string | /etc/cli.sh; <br/>/bin/bash; <br/>/bin/csh; <br/>/bin/tcsh; |
+ |  |  |  |  |  |
+ | **sicKey** | Set the Secure Internal Communication one time secret used to set up trust between the primary and secondary servers. SIC key must be provided if installing a secondary Multi-Domain Server. |
```
Comment (chkp-natanelm) [inline terraform/azure/mds-existing-vnet/README.md:123]: sicKey -> sic_key

File: terraform/azure/mds-existing-vnet/README.md [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+ 
+        terraform plan
+- Create or modify the deployment:
+ 
+        terraform apply
+
+### terraform.tfvars variables:
+ | Name          | Description                                                                                                                                                                          | Type          | Allowed values |
+ |--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------| ------------- | ------------- | -------------  |
+ | **client_secret** | password The client secret of the Service Principal used to deploy the solution                                                                                                      | string |
```
Comment (chkp-natanelm) [inline terraform/azure/mds-existing-vnet/README.md:75]: What is wrong with this description?

File: terraform/azure/mds-existing-vnet/README.md [hunk 8]
```diff
@@ -72,7 +72,7 @@ This solution uses the following modules:
 ### terraform.tfvars variables:
  | Name          | Description                                                                                                                                                                          | Type          | Allowed values |
  |--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------| ------------- | ------------- | -------------  |
- | **client_secret** | password The client secret of the Service Principal used to deploy the solution                                                                                                      | string |
```
Comment (chkp-natanelm) [inline terraform/azure/mds-existing-vnet/README.md]: change to value

**Pull request merged**